VirtualReality/Widgets/qMRMLVirtualRealityHomeWidget.cxx

```diff
@@ -106,6 +113,57 @@ vtkMRMLVirtualRealityViewNode* qMRMLVirtualRealityHomeWidget::virtualRealityView
   return d->VirtualRealityViewNode;
 }
 
+//-----------------------------------------------------------------------------
+void qMRMLVirtualRealityHomeWidget::addModuleButton(QWidget* widget, QIcon* icon)
+{
+  Q_D(qMRMLVirtualRealityHomeWidget);
+
+  QPushButton* moduleButton = new QPushButton(d->ModulesGroupBox);  
+  d->ModulesGroupBoxLayout->addWidget(moduleButton);
+  moduleButton->setIcon(*icon);
+  
+  widget->setParent(d->ModuleWidgetFrame);
+  d->ModuleWidgetsMap[moduleButton] = widget;
+
+  QObject::connect(moduleButton, SIGNAL(clicked()), this, SLOT(onModuleButtonPressed()));
+  QObject::connect(d->backButton, SIGNAL(clicked()), this, SLOT(onBackButtonPressed()));
+ 
+  moduleButton->setVisible(true);
+  widget->setVisible(false);
+  d->backButton->setVisible(false);
+}
+
+//-----------------------------------------------------------------------------
+void qMRMLVirtualRealityHomeWidget::onModuleButtonPressed()
+{
+  Q_D(qMRMLVirtualRealityHomeWidget);
+ 
+  QPushButton* moduleButton = qobject_cast<QPushButton*>(sender());
+  
+  d->HomeWidgetFrame->setVisible(false);
+  d->ModulesGroupBox->setVisible(false);
+  d->ModuleWidgetsMap[moduleButton]->setVisible(true);
+  d->backButton->setVisible(true);
+}
+
+//-----------------------------------------------------------------------------
+void  qMRMLVirtualRealityHomeWidget::onBackButtonPressed()
+{
+  Q_D(qMRMLVirtualRealityHomeWidget);
+  d->HomeWidgetFrame->setVisible(true);
+  d->ModulesGroupBox->setVisible(true); 
+  d->backButton->setVisible(false);
+
+  QMap<QPushButton*, QWidget*>::const_iterator iteratorForMap; 
+  for (iteratorForMap = d->ModuleWidgetsMap.constBegin(); 
```

Comment:
We're not using 80 character terminals any more. Just let the for declaration go in one line. It will be easier to read. Thanks!